src/nwb2bids/_tools/_cache_nwb.py

```diff
@@ -4,15 +4,21 @@
 
 import pynwb
 
-warnings.filterwarnings(action="ignore", message="No cached namespaces found in .+")
-warnings.filterwarnings(action="ignore", message="Ignoring cached namespace .+")
-warnings.filterwarnings(action="ignore", message="Stimulus description 'NA' for .+")
-warnings.filterwarnings(action="ignore", message="Use of icephys_filtering is deprecated and will be removed .+")
+ignored_pynwb_message_patterns = [
+    "No cached namespaces found in .+",
+    "Ignoring cached namespace .+",
+    "Ignoring the following cached .+",
+    "Stimulus description 'NA' for .+",
+    "Use of icephys_filtering is deprecated and will be removed .+",
+    "Date is missing timezone .+",
+]
+for message in ignored_pynwb_message_patterns:
+    warnings.filterwarnings(action="ignore", message=message)
 
 
 @functools.cache
 def cache_read_nwb(file_path: pathlib.Path) -> pynwb.NWBFile:
     """Cache the read operation per NWB file path to speed up repeated calls."""
-
-    nwbfile = pynwb.read_nwb(path=file_path)
+    resolved_path = file_path.resolve()  # Always resolve in case of symlinks (i.e., DataLad most likely)
+    nwbfile = pynwb.read_nwb(path=resolved_path)
     return nwbfile
```

Comment:
why does it matter to resolve symlinks???  make comment describing why, not what it does (obvious)

why bother with temp variable, IMHO should be just 

```suggestion
    return pynwb.read_nwb(path=resolved_path)
```

Reply:
> why does it matter to resolve symlinks???

It matters quite a bit on Windows. That is the action that 'follows' their artificial approach to them. Probably has less relevance to other OS

Added a comment


> why bother with temp variable, IMHO should be just

Again with the usage of temp variables, they make debugging much easier when entering a namespace in debug mode. Saves a lot of time for the people that actually go into these things when fixing tests